### PR TITLE
Update alias on local add and remove

### PR DESCRIPTION
**Pull request by freddygv**
Fixes: #5634 and #5332

Previously alias check updates were only triggered when the local state called `UpdateCheck`. 

When a service is put into `maint` mode, a new check is added when enabled and removed when disabled. Because maintenance doesn't update a check, the alias check update was never triggered.

This PR adds a check for alias notifications to the remaining local state check operations.

### CODE REVIEW

File: agent/local/state.go [hunk 1]
```diff
@@ -488,6 +488,9 @@ func (l *State) addCheckLocked(check *structs.HealthCheck, token string) error {
 		return fmt.Errorf("Check %q refers to non-existent service %q", check.CheckID, check.ServiceID)
 	}
 
+	// If this is a check for an aliased service, then notify the waiters.
+	l.notifyIfAliased(check.ServiceID)
```
Comment (mkeeler) [inline agent/local/state.go]: Wont the call to `setCheckStateLocked` below also call this. Is it necessary to call it in both places?

File: agent/local/state_test.go [hunk 2]
```diff
@@ -2211,6 +2211,98 @@ func TestStateProxyRestore(t *testing.T) {
...
+		Node:      "",
+		CheckID:   proxyID,
+		Name:      "Connect Sidecar Aliasing socat",
+		Status:    api.HealthPassing,
+		ServiceID: scID,
+	}
+	chkt := &structs.CheckType{
+		AliasService: svcID,
+	}
+	if err := a.AddCheck(chk1, chkt, true, "", agent.ConfigSourceLocal); err != nil {
```
Comment (mkeeler) [inline agent/local/state_test.go]: Could be simplified to 

```go
require.NoError(t, a.AddCheck(chk1, chkt, true, "", agent.ConfigureSourceLocal))
```

File: agent/local/state_test.go [hunk 3]
```diff
@@ -2211,6 +2211,98 @@ func TestStateProxyRestore(t *testing.T) {
...
+	chk2 := &structs.HealthCheck{
+		Node:      "",
+		CheckID:   maintID,
+		Name:      "socat:Service Maintenance Mode",
+		Status:    api.HealthCritical,
+		ServiceID: svcID,
+	}
+	a.State.AddCheck(chk2, "")
+
+	retry.Run(t, func(r *retry.R) {
```
Comment (mkeeler) [inline agent/local/state_test.go]: All of these `retry.Runs` can be simplified using the `testify/require` to something like the following.

```go
retry.Run(t, func(r *retry.R) {
   require.Equal(r, api.HealthCritical, a.State.Check(proxyID).Status)
})
```

File: agent/local/state.go (hunk 1 above)
Comment (freddygv) [reply agent/local/state.go]: Nope, it's not necessary. Just removed this one.

**Comment by banks** [comment]
🎉 👏 Great job @freddygv!

**Pull request merged**